proxy/src/source.rs

```diff
@@ -31,21 +32,23 @@ pub struct ProjectId {
     pub domain: String,
 }
 
-impl From<radicle_registry_client::ProjectId> for ProjectId {
-    fn from(id: radicle_registry_client::ProjectId) -> Self {
-        Self {
+impl TryFrom<radicle_registry_client::ProjectId> for ProjectId {
+    type Error = &'static str;
+    fn try_from(id: radicle_registry_client::ProjectId) -> Result<Self, Self::Error> {
+        Ok(Self {
             name: id.0.to_string(),
             domain: id.1.to_string(),
-        }
+        })
     }
 }
 
-impl Into<radicle_registry_client::ProjectId> for ProjectId {
-    fn into(self) -> radicle_registry_client::ProjectId {
-        (
+impl TryInto<radicle_registry_client::ProjectId> for ProjectId {
+    type Error = &'static str;
+    fn try_into(self) -> Result<radicle_registry_client::ProjectId, Self::Error> {
+        Ok((
             ProjectName::from_string(self.name).expect("project name creation failed"),
```

Comment:
This seems like an anti-pattern. You `expect` inside of a thing that returns a `Result`. I'd expect the use of `?` to propagate the error.